idaes/models_extra/co2_capture_and_utilization/unit_models/membrane_1d.py

```diff
@@ -0,0 +1,341 @@
+from idaes.models.unit_models.mscontactor import MSContactor
+from idaes.core.util.exceptions import ConfigurationError, InitializationError
+
+# TODO: add robust initialization
+__author__ = "Maojian Wang"
+
+
+class MembraneFlowPattern(Enum):
+    """
+    Enum of supported flow patterns for membrane.
+    So far only support countercurrent and cocurrent flow
+    """
+
+    countercurrent = 1
+    cocurrent = 2
+    crossflow = 3
+
+
+@declare_process_block_class("Membrane1D")
+class Membrane1DData(UnitModelBlockData):
+    """Standard Membrane 1D Unit Model Class."""
+
+    CONFIG = UnitModelBlockData.CONFIG()
+
+    Stream_Config = ConfigDict()
+
+    Stream_Config.declare(
+        "property_package",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for given stream",
+            doc="""Property parameter object used to define property calculations for given stream,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}""",
+        ),
+    )
+
+    Stream_Config.declare(
+        "property_package_args",
+        ConfigDict(
+            implicit=True,
+            description="Dict of arguments to use for constructing property package",
+            doc="""A ConfigDict with arguments to be passed to property block(s)
+    and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+
+    Stream_Config.declare(
+        "flow_direction",
+        ConfigValue(
+            default=FlowDirection.forward,
+            domain=In(FlowDirection),
+            doc="Direction of flow for stream",
+            description="FlowDirection Enum indicating direction of "
+            "flow for given stream. Default=FlowDirection.forward.",
+        ),
+    )
+
+    Stream_Config.declare(
+        "has_energy_balance",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            doc="Bool indicating whether to include energy balance for stream. Default=True.",
+        ),
+    )
+
+    Stream_Config.declare(
+        "has_pressure_balance",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            doc="Bool indicating whether to include pressure balance for stream. Default=True.",
+        ),
+    )
+
+    Stream_Config.declare(
+        "has_feed",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            doc="Bool indicating whether stream has a feed.",
+            description="Bool indicating whether stream has a feed Port and inlet "
+            "state, or if all flow is provided via mass transfer. Default=True.",
+        ),
+    )
+    CONFIG.declare(
+        "sweep_flow",
```

Comment:
This appears to duplicate the `has_feed` options in the `StreamCONFIG`. Do you actually need both (I suspect you only need this one)?

Reply:
Thanks for pointing that out. You're right. However, I think "has_feed" is too generic. I'll keep "sweep_flow" there to maintain straightforward options.